### PR TITLE
[bitnami/redis-cluster] support external host to cluster announcement preffered type

### LINKED ISSUE #16242: [bitnami/redis-cluster] Cluster init script hangs when external access is enabled in eks cluster using aws classic load balancers
### Name and Version

bitnami/redis-cluster 8.4.4

### What architecture are you using?

None

### What steps will reproduce the bug?

I've been following this guide to enable external access to redis-cluster in an amazon eks cluster: https://docs.bitnami.com/tutorials/deploy-redis-cluster-tmc-helm-chart/#step-5-deploy-the-bitnami-redis-cluster-helm-chart-by-enabling-external-access

The first step completes successfully and I am able to see that the load balancers are created with ports 6379 and 16379 exposed:
```
redis-cluster-jl                  ClusterIP      172.20.48.94     <none>                    6379/TCP                         18h
redis-cluster-jl-0-svc            LoadBalancer   172.20.243.179   *****.elb.amazonaws.com   6379:32112/TCP,16379:30316/TCP   18h
redis-cluster-jl-1-svc            LoadBalancer   172.20.229.93    *****.elb.amazonaws.com   6379:32357/TCP,16379:31750/TCP   18h
redis-cluster-jl-2-svc            LoadBalancer   172.20.101.213   *****.elb.amazonaws.com   6379:32219/TCP,16379:30617/TCP   18h
redis-cluster-jl-3-svc            LoadBalancer   172.20.19.236    *****.elb.amazonaws.com   6379:32177/TCP,16379:32397/TCP   18h
redis-cluster-jl-4-svc            LoadBalancer   172.20.90.157    *****.elb.amazonaws.com   6379:31838/TCP,16379:30064/TCP   18h
redis-cluster-jl-5-svc            LoadBalancer   172.20.224.185   *****.elb.amazonaws.com   6379:30587/TCP,16379:30976/TCP   18h
redis-cluster-jl-headless         ClusterIP      None             <none>                    6379/TCP,16379/TCP               18h
``` 

After adding the load balancer addresses to the `cluster.externalAccess.service.loadbalancerIP` array and performing the helm upgrade, the statefulset is created and after some time, all 6 nodes appear to come up and report healthy:

image

However, after further inspection of the logs on pod `-0` it appears that the cluster init script is hanging on the following message:

```
>>> Sending CLUSTER MEET messages to join the cluster
Waiting for the cluster to join
```

In addition, `cluster info` reports that the cluster status is `fail` and that not all 6 nodes have joined successfully:
```
cluster_state:fail
cluster_slots_assigned:10923
cluster_slots_ok:10923
cluster_slots_pfail:0
cluster_slots_fail:0
cluster_known_nodes:2
cluster_size:2
cluster_current_epoch:2
cluster_my_epoch:2
cluster_stats_messages_ping_sent:9005
cluster_stats_messages_meet_sent:1
cluster_stats_messages_sent:9006
cluster_stats_messages_pong_received:2336
cluster_stats_messages_received:2336
total_cluster_links_buffer_limit_exceeded:0
```

I have engaged aws cloud support to rule out connectivity issues, and we were able to successfully telnet into all of the loadbalancers on port `6379` as well as `16379` from a test pod within the k8s cluster.

### Are you using any custom parameters or values?

```
cluster:
  externalAccess:
    enabled: true
    service:
      type: LoadBalancer
    loadbalancerIP:
    - load-balancer-0-ip
    - load-balancer-1-ip
    - load-balancer-2-ip
    - load-balancer-3-ip
    - load-balancer-4-ip
    - load-balancer-5-ip
```

### What is the expected behavior?

The cluster boots up successfully and reports that the cluster status is ok. All 6 nodes are joined to the cluster and external clients are able to connect, auth, read and write keys.

### What do you see instead?

Cluster init script hangs on the message:

```
>>> Sending CLUSTER MEET messages to join the cluster
Waiting for the cluster to join
```

And `cluster info` reports that the cluster status is `fail`. Not all 6 nodes have joined successfully:
```
cluster_state:fail
cluster_slots_assigned:10923
cluster_slots_ok:10923
cluster_slots_pfail:0
cluster_slots_fail:0
cluster_known_nodes:2
cluster_size:2
cluster_current_epoch:2
cluster_my_epoch:2
cluster_stats_messages_ping_sent:9005
cluster_stats_messages_meet_sent:1
cluster_stats_messages_sent:9006
cluster_stats_messages_pong_received:2336
cluster_stats_messages_received:2336
total_cluster_links_buffer_limit_exceeded:0
```

### Additional information

EKS cluster k8s version - 1.24.10
AWS load balancer controller version - 2.4.5

### PR DESCRIPTION
**Pull request by yo-ga**
### Description of the change

Support the external access host mode integrated with external load balancer in redis cluster, such as AWS LoadBalancer.

### Benefits

Operator can set and use the load balancer hostname directly with redis cluster mode.

### Possible drawbacks

NA

### Applicable issues

- fixes #16242 

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by yo-ga** [comment]
@carrodher  Is there any updates or reviews?

File: bitnami/redis-cluster/Chart.yaml [hunk 1]
```diff
@@ -25,4 +25,4 @@ maintainers:
 name: redis-cluster
 sources:
   - https://github.com/bitnami/charts/tree/main/bitnami/redis-cluster
-version: 8.6.12
+version: 8.6.13
```
Comment (aoterolorenzo) [inline bitnami/redis-cluster/Chart.yaml:28]: I would update a minor version since dis adds new support

**Review by aoterolorenzo — CHANGES_REQUESTED** [review]
Hi @yo-ga,

Sorry for all the delay with this, and thanks for the contribution!

I just left a minor comment, for the rest LGTM

Could you update the branch to merge? Thank you.

**Review by aoterolorenzo — APPROVED** [review]
LGTM

**Pull request merged**